compiler/resolution/src/org/jetbrains/kotlin/resolve/calls/tower/ScopeTowerProcessors.kt

```diff
@@ -149,36 +140,16 @@ private class NoExplicitReceiverScopeTowerProcessor<C : Candidate>(
     val collectCandidates: CandidatesCollector
 ) : AbstractSimpleScopeTowerProcessor<C>(context) {
     override fun simpleProcess(data: TowerData): Collection<C> = when (data) {
-        is TowerData.TowerLevel -> {
-            val result = mutableListOf<C>()
-            for (towerCandidate in data.level.collectCandidates(null)) {
-                if (!towerCandidate.requiresExtensionReceiver) {
-                    result.add(
-                        candidateFactory.createCandidate(
-                            towerCandidate,
-                            ExplicitReceiverKind.NO_EXPLICIT_RECEIVER,
-                            extensionReceiver = null
-                        )
-                    )
-                }
-            }
-            result
-        }
-        is TowerData.BothTowerLevelAndImplicitReceiver -> {
-            val result = mutableListOf<C>()
-            for (towerCandidate in data.level.collectCandidates(data.implicitReceiver)) {
-                if (towerCandidate.requiresExtensionReceiver) {
-                    result.add(
-                        candidateFactory.createCandidate(
-                            towerCandidate,
-                            ExplicitReceiverKind.NO_EXPLICIT_RECEIVER,
-                            extensionReceiver = data.implicitReceiver
-                        )
-                    )
-                }
-            }
-            result
-        }
+        is TowerData.TowerLevel -> resolve(
+            data.level.collectCandidates(null),
+            { !it },
+            ExplicitReceiverKind.NO_EXPLICIT_RECEIVER,
+            null)
```

Comment:
`)` is supposed to be placed on the next line.
The same applies to the when entry below